src/backend/main.py

```
@@ -6,91 +7,80 @@
+        pr_response.raise_for_status()
+        
+        files_response = requests.get(files_url, headers=headers)
+        files_response.raise_for_status()
+        
+        pr_data = pr_response.json()
+        pr_data['files'] = files_response.json()
+        
+        return pr_data
+        
+    except requests.exceptions.RequestException as e:
+        raise GitHubAuthError(f"Failed to fetch PR details: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)